samples/gitchurn/src/gitChurnMain/kotlin/GitChurn.kt

```diff
@@ -35,33 +35,44 @@ fun main(args: Array<String>) {
 private fun calculateChurn(workDir: String, limit: Int) {
     println("Opening…")
     val repository = git.repository(workDir)
-    val map = mutableMapOf<String, Int>()
+    data class WhoAndWhere(val authorEmail: String = "", val filePath: String = "")
+    val modificationsByAuthor = mutableMapOf<WhoAndWhere, Int>()
     var count = 0
-    val commits = repository.commits()
-    val limited = commits.take(limit)
+    val commits = repository.myCommits()
+    val commitsLimit = commits.take(limit)
     println("Calculating…")
-    limited.forEach { commit ->
+    commitsLimit.forEach { commit ->
         if (count % 100 == 0)
-            println("Commit #$count [${commit.time.format()}]: ${commit.summary}")
+            println("Commit #$count [${commit.time.format()}] by ${commit.author.name!!.toKString()}: ${commit.summary}")
 
         commit.parents.forEach { parent ->
             val diff = commit.tree.diff(parent.tree)
             diff.deltas().forEach { delta ->
-                val path = delta.newPath
-                val n = map[path] ?: 0
-                map.put(path, n + 1)
+                val path = if (delta.newPath == delta.newPath.substringBefore('/'))
```

Comment:
Do I understand correctly that this code computes the path to the directory containing the file?
Somewhat unreadable.
Does it handle files located at the root directory properly?

Reply:
Not exactly, this code idea was to find out, which of the root directories this exact file lies in. It was a move towards "where this person was working" analysis, that differs from original "what files were touched what number of times". And yes, it handles files from the root, in the report they can be spotted by missing '/' in the end.

Reply:
> which of the root directories this exact file lies in

Are there multiple root directories?

> And yes, it handles files from the root, in the report they can be spotted by missing '/' in the end.

Does this code produce file paths for files from root directory and containing directory paths for others?

Reply:
I meant all directories, that are inside of the root one.
> Does this code produce file paths for files from root directory and containing directory paths for others?
Yes, but only highest-level part of the directory path, if I use this term the right way.

Reply:
Now I see. Thanks for the explanation.
I believe it should be commented with a couple of examples, e.g. 
```
// foo/bar/baz.txt -> foo/
// foo.txt -> foo.txt
```

Also `delta.newPath == delta.newPath.substringBefore('/')` looks equivalent to `!delta.newPath.contains('/')`. If this is true, then simpler variant is preferable.